class/defaults.yml

```
@@ -2,3 +2,17 @@ parameters:
   appuio_cloud:
     =_metadata: {}
     namespace: syn-appuio-cloud
+
+    reservedNamespaces:
+      allowedClusterRoles:
+        create: true
+        name: appuio:edit-reserved-namespaces
+        additionalRoles:
+          - cluster-admin
+      match:
+        - default
+        - kube-*
+        - openshift-*
+        - syn-*
+        - vshn-*
+        - appuio-*
```

The problem with arrays is that they cannot be easily overridden through Parameter hierarchy. Since arrays are not mergeable, one has to copy-paste everything from the defaults if only one additional entry is required.

Maybe something like this makes more sense:
```suggestion
      match:
        default: {}
        "kube-*": {} # not sure if that's actually valid YAML...
        "openshift-*": {}
        "syn-*": {}
        "vshn-*": {}
        "appuio-*": {}
```
Alternatively, we might need to do it more explicitly:
```yaml
match:
  kube:
    regex: "kube-*"
```

---

I like the second proposal better. 

It makes it easy to remove a match without knowing commodore internals.

I named the key `pattern` since it is not a regex, but a glob pattern.

```yaml
match:
  kube: nil
# or
match:
  kube:
    pattern: nil
```

---

Another alternative would be to make group of arrays, e.g. 
```yaml
match:
  defaultSet:
    - kube-*
    - etc...
  myCustomSet:
    - mycompany-*
```
So to add patterns, give them a name. However, if you want to allow `kube-*`, then you'd have to copy/paste the `defaultSet` to exclude it from the list...

---

Looks way cleaner 🤔 I guess removing restrictions won't be needed that much. So this proposal is more comfortable.